src/features/collapsible-sidebar/__tests__/CollapsibleSidebarMenuItem.test.js

```diff
@@ -32,6 +32,35 @@ describe('components/core/collapsible-sidebar/__tests__/CollapsibleSidebarMenuIt
         expect(wrapper).toMatchSnapshot();
     });
 
+    test('should show custom content when content is passed but not the text', () => {
+        libDom.useIsContentOverflowed.mockReturnValue(false);
+
+        const wrapper = getWrapper(
+            {
+                className: 'foo',
+                content: <div>Custom Content</div>,
+                icon: 'bold',
+            },
+            { isScrolling: false },
+        );
+        expect(wrapper).toMatchSnapshot();
```

Comment:
for the actual check, we should check for the presence of the content instead of relying on snapshots.